pkg/binding/bindingpolicy-resolution.go

```diff
@@ -36,36 +37,62 @@ import (
 type bindingPolicyResolution struct {
 	sync.RWMutex
 
+	// This map is mutable, but every `ObjectData` stored in it is immutable.
+	// No `*ObjectData` in this map is nil.
 	objectIdentifierToData map[util.ObjectIdentifier]*ObjectData
 
 	// Every Set ever stored here is immutable from the time it is stored here.
 	destinations sets.Set[string]
 
 	// ownerReference identifies the bindingpolicy that this resolution is
 	// associated with as an owning object.
+	// This pointer is never nil (why is it a pointer?).
+	// The `metav1.OwnerReference` is immutable.
 	ownerReference *metav1.OwnerReference
 
 	// requiresSingletonReportedState indicates whether the bindingpolicy
 	// that this resolution is associated with requires singleton status.
 	requiresSingletonReportedState bool
 }
 
-// ObjectData stores the UID, resource version, create-only bit,
-// and statuscollectors for an object.
+// ObjectData stores most of the downsync modalities for a given workload object
+// with respect to a given Binding.
 type ObjectData struct {
-	// UID is the UID of the object.
+	// UID is the UID of the workload object.
 	UID string
-	// ResourceVersion is the resource version of the object.
+	// ResourceVersion is the resource version of the workload object.
 	ResourceVersion string
 	// CreateOnly is a boolean that indicates whether the object was selected
 	// by at least one downsync clause with the createOnly bit set.
 	// If true, it means that the object should only be created and not
 	// maintained
 	CreateOnly bool
-	// StatusCollectors is a set of status collector names that are associated
+	// StatusCollectors is a set of status collector names that are associated.
+	// Every set ever stored here is immutable.
 	StatusCollectors sets.Set[string]
 }
 
+var _ Resolution = &bindingPolicyResolution{}
+
+func (resolution *bindingPolicyResolution) GetPolicyUID() string {
+	resolution.RLock()
+	defer resolution.RUnlock()
+	return string(resolution.ownerReference.UID)
+}
+
+func (resolution *bindingPolicyResolution) GetDestinations() sets.Set[string] {
+	resolution.RLock()
+	defer resolution.RUnlock()
+	return resolution.destinations
+}
+
+func (resolution *bindingPolicyResolution) GetWorkload() abstract.Map[util.ObjectIdentifier, ObjectData] {
+	m1 := abstract.AsPrimitiveMap(resolution.objectIdentifierToData)
+	m2 := abstract.NewMapLocker(&resolution.RWMutex, m1)
+	m3 := abstract.MapMapValues(m2, NonNilPointerDeference)
```

Comment:
These three lines seem very similar to functionalities in pkg/primitive-maps.go. It would be awkward to keep both around.

Reply:
I assume you mean the file in this package that this PR renames to `primitive-map-ops.go`. Yes, there is partial overlap. Yes, I think that overlap should be addressed in follow-on PRs.